src/AbstractActionReader.ts

```diff
@@ -27,10 +28,10 @@ const defaultBlock: Block = {
 /**
  * Reads blocks from a blockchain, outputting normalized `Block` objects.
  */
-export abstract class AbstractActionReader {
-  public startAtBlock: number
-  public headBlockNumber: number = 0
-  public currentBlockNumber: number
+export abstract class AbstractActionReader implements ActionReader {
+  protected startAtBlock: number
+  protected headBlockNumber: number = 0
+  protected currentBlockNumber: number
```

Comment:
Making these protected will break BaseActionWatcher, which uses a direct reference here.
https://github.com/EOSIO/demux-js/blob/develop/src/BaseActionWatcher.ts#L147

It also might break custom implementations some projects might have, since we're changing the public API in this case.

Reply:
You can access these from the `info` property, which is required by the `ActionReader` interface.

And yes, this is a breaking change- there will be an upgrade guide the next time we make a release.